Clang in MSVC mode broken in 2.4.9

## Description
After updating doctest to 2.4.9, it broke clang in msvc mode again. e8ba771d4b4cf90cb99e011ef01059dd6533d39c introduced some changes, using `::operator<<` in stringification, possibly to workaround msvc bugs, but it causes bugs in clang instead:
```
doctest.h:978:56: error: no member named 'operator<<' in the global namespace
    struct has_global_insertion_operator<T, decltype(::operator<<(declval<std::ostream&>(), declval<const T&>()), void())> : types::true_type { };
                                                     ~~^
1 error generated.
```

https://github.com/doctest/doctest/blob/b7c21ec5ceeadb4951b00396fc1e4642dd347e5f/doctest/doctest.h#L973
https://github.com/doctest/doctest/blob/b7c21ec5ceeadb4951b00396fc1e4642dd347e5f/doctest/doctest.h#L1124
Replacing these `#if`s with `#if !DOCTEST_CLANG && (defined(_MSC_VER) && _MSC_VER <= 1900)` seem to fix the problem for me.

### Steps to reproduce
Just include the doctest header.

### Extra information
* doctest version: **v2.4.9**
* Operating System: **Gentoo linux**
* Compiler+version: **clang 8.0.1 with msvc12 stdlibs**

--- a/doctest/doctest.h
+++ b/doctest/doctest.h
@@ -979,7 +979,7 @@ namespace detail {
     struct deferred_false : types::false_type { };
 
 // MSVS 2015 :(
-#if defined(_MSC_VER) && _MSC_VER <= 1900
+#if !DOCTEST_CLANG && defined(_MSC_VER) && _MSC_VER <= 1900
     template <typename T, typename = void>
     struct has_global_insertion_operator : types::false_type { };
 

--- a/doctest/parts/doctest_fwd.h
+++ b/doctest/parts/doctest_fwd.h
@@ -976,7 +976,7 @@ namespace detail {
     struct deferred_false : types::false_type { };
 
 // MSVS 2015 :(
-#if defined(_MSC_VER) && _MSC_VER <= 1900
+#if !DOCTEST_CLANG && defined(_MSC_VER) && _MSC_VER <= 1900
     template <typename T, typename = void>
     struct has_global_insertion_operator : types::false_type { };
 

